File: Week 1/Cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int cents;

    // Prompt user until they give a non-negative integer
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    int coins = 0;

    // Quarters
    coins += cents / 25;
    cents = cents % 25;

    // Dimes
    coins += cents / 10;
    cents = cents % 10;

    // Nickels
    coins += cents / 5;
    cents = cents % 5;

    // Pennies
    coins += cents; // remaining cents are all pennies

    printf("%d\n", coins);
}

File: Week 6/Sentimental - cash/cash.py
while True:
    try:
        change = float(input("Change owed: "))
        if change >= 0:
            break
    except ValueError:
        pass

# Convert dollars to cents, rounding to avoid floating-point issues
cents = round(change * 100)

coins = 0

# Quarters
coins += cents // 25
cents %= 25

# Dimes
coins += cents // 10
cents %= 10

# Nickels
coins += cents // 5
cents %= 5

# Pennies
coins += cents

print(coins)
